src/NLog/Internal/NetworkSenders/HttpNetworkSender.cs

```diff
@@ -80,16 +77,17 @@ protected override void DoSend(byte[] bytes, int offset, int length, AsyncContin
                         }
 
                         // completed fine
-                        asyncContinuation(null);
+                        base.EndRequest(asyncContinuation, null);
                     }
                     catch (Exception ex)
                     {
+                        InternalLogger.Error(ex, "NetworkTarget: Error sending http request");
                         if (ex.MustBeRethrownImmediately())
                         {
                             throw; // Throwing exceptions here will crash the entire application (.NET 2.0 behavior)
                         }
 
-                        asyncContinuation(ex);
+                        base.EndRequest((e) => asyncContinuation(ex), null);    // pendingException = null to keep sender alive
```

Comment:
Is it correct we're ignoring `e`. Maybe use `_` otherwise?

Reply:
Fixed